asm/macros/battle_script.inc

```diff
@@ -2177,6 +2171,12 @@
 	jumpifbyte CMP_COMMON_BITS, gMoveResultFlags, MOVE_RESULT_NO_EFFECT, \jumpInstr
 	.endm
 
+	.macro jumpifside battler:req, side:req, equalJumpInstr:req, notEqualJumpInstr:req
+	getbattlerside \battler
+	jumpifbyte CMP_EQUAL, gBattleCommunication, \side, \equalJumpInstr
+	jumpifbyte CMP_NOT_EQUAL, gBattleCommunication, \side, \notEqualJumpInstr
+	.endm
```

Comment:
Uuh, aren't the jumpif macros meant to let the script continue to the next instruction if the condition doesn't pass?

Reply:
They're meant to be used however the situation requires it. It's no different than doing a `call_if_eq` followed by a `call_if_ne` inside an overworld script.

In this case, the macro `jumpifside` first calls `getbattlerside` to write the side that the specified battler is on to `gBattleCommunication`, and then proceeds to jump to one battle script or another based on its value *(which is either `B_SIDE_PLAYER` or `B_SIDE_OPPONENT`)*.

Reply:
Yes, but the `goto BattleScript_ButItFailed` is never reached, so the second condition might as well be converted to:
```diff
-	jumpifside BS_ATTACKER, B_SIDE_PLAYER, BattleScript_EffectBatonPass, BattleScript_EffectTeleportTryToRunAway
-	goto BattleScript_ButItFailed
+	jumpifside BS_ATTACKER, B_SIDE_PLAYER, BattleScript_EffectBatonPass
+	goto BattleScript_EffectTeleportTryToRunAway
```

Reply:
Applied

Reply:
Wait a second, I just realized that `BattleScript_EffectTeleportTryToRunAway` is right below it, so you can just remove that goto ^^;
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/1226b8c4-f7e3-4b7c-93ac-3a21c1e3a548)